### ISSUE TITLE
Unexpected null behaviour

### ISSUE THREAD

**Issue by TimothyMakkison** [Cysharp/SimdLinq#2]
It appears that unlike Linq, the extension methods do not check for null arrays or lists. Should a null check be added? If this is intentional then the readme should probably mention this behaviour.

``` C#
 int[] ints = null;
 // Returns 0
 var response = SimdLinqExtensions.Sum(ints);
// Throws ArgumentNullException
var linqResponse = Enumerable.Sum(ints);
```

**Comment by neuecc** [comment]
thanks, requires null check.
I'll release new version soon.

**Comment by neuecc** [comment]
I've released 1.1.0, it checks null.
